src/Sticky.jsx

```diff
@@ -194,75 +198,78 @@ class Sticky extends Component {
 
     /**
      * Update Sticky position.
+     * @param {Object} state optional value for this.state in case setState hasn't landed yet
      */
-    update () {
+    update (state) {
```

Comment:
I'm not a fan of passing state around like this, as it seems to add complexity and increases the risk of things getting out of sync.  Instead, it might make more sense to pass a callback to `updateInitialDimension()`, which just calls `update()` which can be passed as the callback to `setState()` when it's called inside `updateInitialDimension()`.  That way you can be sure that when `update()` is eventually executed, `this.state` will always reflect the current state.